### PR TITLE
fix: add missing properties for node http.ClientRequest

### PR DESCRIPTION
**Pull request by twchn**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://nodejs.org/api/http.html#class-httpclientrequest>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@twchn thx!
the only nitpick is, should 'reusedSocket' be readonly(it's not supposed to be set), but there is nothing like strict convention here for this IMO in Node types.

**Comment by twchn** [comment]
@peterblazejewicz Thanks for your review, perhaps it should be consistent with other properties currently, is it better to have another PR(may be verison breaking) to fix all readonly properties yet?

File: types/node/http.d.ts [hunk 1]
```diff
@@ -621,6 +621,15 @@ declare module 'http' {
          * @since v14.5.0, v12.19.0
          */
         protocol: string;
+        /**
+         * Whether the request is send through a reused socket.
+         * @since v13.0.0, v12.16.0
+         */
+        reusedSocket: boolean;
+        /**
+         * Limits maximum response headers count. If set to 0, no limit will be applied. Default = 2000.
```
Comment (LinusU) [inline types/node/http.d.ts:630]: Potentially we could use the [`@default`](https://jsdoc.app/tags-default.html) tag here?

```suggestion
         * Limits maximum response headers count. If set to 0, no limit will be applied.
         * @default 2000
```

**Review by LinusU — APPROVED** [review]
Nice 👍

**Review by peterblazejewicz — APPROVED** [review]

**Comment by SimonSchick** [comment]
@twchn we generally don't minor update creep in PRs, feel free to change it to readonly, not required though.

**Review by peterblazejewicz — APPROVED** [review]

**Comment by twchn** [comment]
Ready to merge

**Pull request merged**